tests/epochs.test.mjs

```diff
@@ -8,148 +8,337 @@ import {
+            compositeWeight: 4,
             gatewayAddress: STUB_OPERATOR_ADDRESS,
-            stakeWeight: 1,
             gatewayPerformanceRatio: 1,
-            observerPerformanceRatio: 1,
-            compositeWeight: 4,
             normalizedCompositeWeight: 1,
-            tenureWeight: 4,
+            observerAddress: STUB_ADDRESS,
+            observerPerformanceRatio: 1,
+            stakeWeight: 1,
             stake: INITIAL_OPERATOR_STAKE,
-            startTimestamp: STUB_TIMESTAMP,
-          },
-        ],
-        prescribedNames: ['prescribed-name'],
-        observations: {
-          failureSummaries: [],
-          reports: [],
-        },
-        distributions: {
-          totalEligibleGatewayReward: 22500900000,
-          totalEligibleGateways: 1,
-          totalEligibleObserverReward: 2500100000,
-          totalEligibleRewards: 25001000000,
-          rewards: {
-            eligible: {
-              [STUB_OPERATOR_ADDRESS]: {
-                delegateRewards: [],
-                operatorReward: 25001000000, // 0.001 of the protocol balance after the transfers and name purchase
-              },
-            },
+            startTimestamp: 0,
+            tenureWeight: 4,
           },
-        },
+        ]);
       });
 
-      // TODO (PE-7321): add a test for an empty epoch before names and gateways have been prescribed
-    });
-  });
-
-  describe('Prescribed Observers', () => {
-    it('should return the correct epoch for the current epoch with weights', async () => {
-      const prescribedObservers = await getPrescribedObservers({
-        memory: sharedMemory,
-        timestamp: firstEpochStartTimestamp,
+      it('should return the prescribed names once the epoch has been prescribed', async () => {
+        const prescribedNames = await getPrescribedNames({
+          memory: sharedMemory,
+          timestamp:
+            epochSettings.epochZeroStartTimestamp +
+            epochSettings.distributionDelayMs,
+        });
+        assert.deepStrictEqual(prescribedNames, ['prescribed-name']);
       });
-      assert.deepStrictEqual(prescribedObservers, [
-        {
-          compositeWeight: 4,
-          gatewayAddress: STUB_OPERATOR_ADDRESS,
-          gatewayPerformanceRatio: 1,
-          normalizedCompositeWeight: 1,
-          observerAddress: STUB_ADDRESS,
-          observerPerformanceRatio: 1,
-          stakeWeight: 1,
-          stake: INITIAL_OPERATOR_STAKE,
-          startTimestamp: STUB_TIMESTAMP,
-          tenureWeight: 4,
-        },
-      ]);
     });
-  });
 
-  describe('Prescribed Names', () => {
-    it('should return the correct epoch for the first epoch', async () => {
-      const prescribedNames = await getPrescribedNames({
-        memory: sharedMemory,
-        timestamp: firstEpochStartTimestamp,
+    describe('Create Next Epoch', () => {
+      it('should create the next epoch at the next epoch start timestamp', async () => {
```

Comment:
new test to validate the creation of the next epoch can occur exactly at its start timestamp